backend/compact-connect/common_constructs/python_function.py

```diff
@@ -72,19 +75,22 @@ def __init__(
                 },
             ],
         )
-        NagSuppressions.add_resource_suppressions_by_path(
-            stack,
-            path=f'{self.node.path}/ServiceRole/Resource',
-            suppressions=[
-                {
-                    'id': 'AwsSolutions-IAM4',
-                    'applies_to': [
-                        'Policy::arn:<AWS::Partition>:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole',
-                    ],
-                    'reason': 'The BasicExecutionRole policy is appropriate for these lambdas',
-                },
-            ],
-        )
+
+        # We only need to suppress this if a role is being created for this lambda
```

Comment:
Do you mean to say, if a role is _not_ being created for this lambda? (I am assuming so since the check is that `role is None`.)

Reply:
Oh, I can see that this would be confusing. In the case that `role` is `None`, a new role will be created for this particular lambda on construction. If `role` is not `None`, that means that a role is being provided from elsewhere, rather than created, in which case, any Nag suppression would need to be handled wherever the role was created.

Reply:
Ah ok. If this understanding is correct, maybe we can update the comment to say "If We only need to suppress this if a role is being created for this lambda

```suggestion
        # Since this lambda won't be assigned a custom role, it will automatically create one with a BasicExecutionRole policy.
```